scrapers_library/data_portals/muckrock/muckrock_tools/search_foia_data_db.py

```diff
@@ -0,0 +1,69 @@
+import sqlite3
+import pandas as pd
+import json
+import argparse
+
+
+def parse_communications_column(communications):
+    if pd.isna(communications):
+        return []
+    try:
+        return json.loads(communications)
+    except json.JSONDecodeError as e:
+        print(f"Error decoding JSON: {e}")
+        return []
+
+
+def search_foia_db(search_string):
+    print(f"'Searching foia_data.db for '{search_string}'...")
+
+    try:
+        with sqlite3.connect('foia_data.db') as conn:
+            query = '''
+                SELECT * FROM results
+                WHERE (title LIKE ? OR tags LIKE ?)
+                AND (status = 'done')
+                '''
+            params = [f'%{search_string}%', f'%{search_string}%']
```

Comment:
This code appears to be able to run when the `results` table does not yet exist! With that in mind, we'd want to either modify this so that `results` is guaranteed created when this is called, or provide a more descriptive error when this happens (as it did to me), indicating that this error is expected and we should do something else before running this.

Reply:
The only scenario I can think of where the db would exist but not the table is if someone creates their own file called `foia_data.db`. There may be a subtle case or two of the db being created programmatically with a corrupted table name though. In any case, I've added checking for not only the existence of the db but also the `results` table for robustness!